src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -12,6 +12,22 @@ class ConfigurationMerger {
         return merged
     }
 
+    @NonCPS
+    def static merge(Map configs, Map configKeys, Map defaults = [:]) {
```

Comment:
The other ```merge```method provided above should redirect to this merge method. IMO it does not make sense to have a merge method with and another one without deep merging.

Reply:
There are significant overlaps with PR #66. This makes the review process harder than necessary.

Reply:
sorry, try to ignore all ConfigMerger files :)

Reply:
Would it be possible to implement the 'ignore' by a corresponding ```rebase -i```